ui/test/unit/appointments/controllers/manage/list/appointmentsListViewController.spec.js

```diff
@@ -91,6 +91,13 @@ describe('AppointmentsListViewController', function () {
         expect(spinner.forPromise).not.toHaveBeenCalled();
     });
 
+    it('should not fetch appointments when doFetchAppointmentsData is set to false', function () {
+        $state.params = {doFetchAppointmentsData: false};
+        createController();
```

Comment:
Why not call
```javascript
scope.getAppointmentsForDate(viewDate);
```
and check that the service was in fact never invoked?

Does `createController()` just provoke those calls?

(Same remark for the ad-hoc test in the other controller.)

Reply:
scope.getAppointmentsForDate(viewDate) is called no matter what the value of the flag variable is.Its just that if the flag is true appointmentsService.getAllAppointments() is executed(this is the API call).If flag is false it will still go into scope.getAppointmentsForDate() ,but not call the API

Reply:
>`scope.getAppointmentsForDate(viewDate)` is called no matter what the value of the flag variable is.

I get that, but (one of) my questions was: is this method ever triggered by your test? Does `createController()` provoke `scope.getAppointmentsForDate(viewDate)`?
If not then the test verifies something obvious.

I was assuming that not, in which case I _think_ your test should invoke `scope.getAppointmentsForDate(viewDate)` and verify that the service was never used.

Reply:
You are correct.I will take a look into it